src/core/modules/Planet.cpp

```diff
@@ -907,12 +1009,66 @@ QString Planet::getInfoStringExtra(const StelCore *core, const InfoStringGroup&
 			{
 				oss << QString("%1: %2%").arg(q_("Eclipse obscuration")).arg(QString::number(eclipseObscuration, 'f', 2)) << "<br />";
 				PlanetP obj = eclObj.second;
-				if (onEarth && obj==ssystem->getMoon())
+				if (onEarth && obj == ssystem->getMoon())
+				{
+					const double eclipseMagnitude =
+					  (0.5 * angularSize
+						+ (obj->getAngularSize(core) * M_PI_180) / obj->getInfoMap(core)["scale"].toDouble()
+						- getJ2000EquatorialPos(core).angle(obj->getJ2000EquatorialPos(core)))
+					  / angularSize;
+					oss << QString("%1: %2")
+							 .arg(q_("Eclipse magnitude"))
+							 .arg(QString::number(eclipseMagnitude, 'f', 3))
+						<< "<br />";
+				}
+			}
+		}
+
+		if (englishName == "Sun" && onEarth)
+		{
+			// Solar eclipse information
+			// Use geocentric coordinates
+			StelCore* core1 = StelApp::getInstance().getCore();
+			const bool useTopocentric = core1->getUseTopocentricCoordinates();
+			core1->setUseTopocentricCoordinates(false);
+			core1->update(0);
+
+			double raSun, deSun, raMoon, deMoon;
+			StelUtils::rectToSphe(&raSun, &deSun, ssystem->getSun()->getEquinoxEquatorialPos(core1));
+			StelUtils::rectToSphe(&raMoon, &deMoon, ssystem->getMoon()->getEquinoxEquatorialPos(core1));
+
+			// R.A. of the Sun
+			raSun = raSun / M_PI_180;
+			if (raSun < 0.) raSun += 360.;
+			// R.A. of the Moon
+			raMoon = (raMoon / M_PI_180);
+			if (raMoon < 0.) raMoon += 360.;
+
+			double raDiff = raMoon - raSun;
+			if (raDiff < 0.) raDiff += 360.;
+
+			if (raDiff < 3. || raDiff > 357.)
+			{
+				SolarEclipse center;
+				Vec3d pos = center.point(core1);
+
+				if (pos[0] < 90.) // only display when shadow axis is touching Earth
 				{
-					const double eclipseMagnitude = (0.5*angularSize + (obj->getAngularSize(core)*M_PI_180)/obj->getInfoMap(core)["scale"].toDouble() - getJ2000EquatorialPos(core).angle(obj->getJ2000EquatorialPos(core)))/angularSize;
-					oss << QString("%1: %2").arg(q_("Eclipse magnitude")).arg(QString::number(eclipseMagnitude, 'f', 3)) << "<br />";
+					oss << QString("%1: %2%3/%4%5")
+							 .arg(q_("Center of solar eclipse (Lat./Long.)"))
+							 .arg(pos[0], 5, 'f', 4)
+							 .arg(QChar(0x00B0))
+							 .arg(pos[1], 5, 'f', 4)
+							 .arg(QChar(0x00B0));
+					oss << "<br/>";
```

Comment:
You could add great-circle distance to observer. I thought we had it implemented already, but now I see only StelLocation::distanceDegrees(). We could add double StelLocation::distanceKm(long1, lat1, long2, lat2), following Meeus, Astron. Algorithms (2nd ed 1998) ch.11, p.85. If you don't have the book, I can do it.

Reply:
I just added such a function to StelLocation. It may even be interesting to say: "Shadow centerpoint is 45km towards azimuth 63°". The azimuth function can be taken from ArchaeoLines::getAzimuthForLocation().

Reply:
@gzotti ArchaeoLines may be not loaded ;) Maybe move getAzimuthForLocation() into StelLocation class?

Reply:
Yes, I meant not "use Archaeoline's function", but "grab code". Indeed, when used elsewhere, this function should be moved, e.g. to StelLocation or StelUtils. 
@worachate001 I am sorry I also implemented the missing bit, now we have a conflict. My implementation is more general, so I'd prefer to keep this. You can move the static functions Archaeolines::getAzimuthForLocation() as Alexander suggested to StelLocation, adapt ArchaeoLines accordingly, and maybe add the non-static function StelLocation::AzimuthForLocation(). I think these make interesting additions!